hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -62,7 +70,51 @@ public void handle(@NonNull final HandleContext context) throws HandleException
     @NonNull
     @Override
     public Fees calculateFees(final FeeContext feeContext) {
-        // Todo: Implement the logic for calculating fees
-        return Fees.FREE;
+        final var body = feeContext.body();
+        final var op = body.tokenRejectOrThrow();
+        final var config = feeContext.configuration();
+        final var tokenMultiplier =
+                requireNonNull(config).getConfigData(FeesConfig.class).tokenTransferUsageMultiplier();
+
+        final int totalRejections = op.rejections().size();
+        int numOfFungibleTokenRejections = 0;
+        int numOfNFTRejections = 0;
+        for (final var rejection : op.rejections()) {
+            if (rejection.hasFungibleToken()) {
+                numOfFungibleTokenRejections++;
+            } else {
+                numOfNFTRejections++;
+            }
+        }
+
+        final int weightedTokensInvolved = tokenMultiplier * totalRejections;
+        final int weightedFungibleTokens = tokenMultiplier * numOfFungibleTokenRejections;
+
+        final long bpt =
+                calculateBytesPerTransaction(weightedTokensInvolved, weightedFungibleTokens, numOfNFTRejections);
+        final long rbs = calculateRamByteSeconds(weightedTokensInvolved, weightedFungibleTokens, numOfNFTRejections);
+
+        return feeContext
+                .feeCalculator(getSubType(numOfNFTRejections))
+                .addBytesPerTransaction(bpt)
+                .addRamByteSeconds(rbs * USAGE_PROPERTIES.legacyReceiptStorageSecs())
+                .calculate();
+    }
+
+    private SubType getSubType(final int numOfNFTRejections) {
+        return numOfNFTRejections != 0 ? TOKEN_NON_FUNGIBLE_UNIQUE : TOKEN_FUNGIBLE_COMMON;
```

Comment:
In CryptoTransferHandler, we determine the SubType in a similar way. We first check if there is a NFT transfer, if not, only then we have FUNGIBLE. 
This means that if there are both Fungible and NFT tokens to be rejected, the subtype would be NON_FUNGIBLE_TOKEN.